include/picongpu/fields/incidentField/profiles/PlaneWave.hpp

```diff
@@ -42,122 +42,106 @@ namespace picongpu
                          *                  fieldE_internal = fieldE_SI / unitField
                          */
                         HINLINE PlaneWaveFunctorIncidentE(float_X const currentStep, float3_64 const unitField)
-                            : Base(unitField)
-                            , elong(getLongitudinal(currentStep))
+                            : Base(currentStep, unitField)
                         {
                         }
 
                         /** Calculate incident field E value for the given position
                          *
-                         * Since it is a plane wave parallel to the generating surface, the value only depends on time.
+                         * Interface required by Base.
                          *
                          * @param totalCellIdx cell index in the total domain (including all moving window slides)
                          * @return incident field E value in internal units
                          */
                         HDINLINE float3_X operator()(floatD_X const& totalCellIdx) const
                         {
-                            return elong;
+                            return Base::operator()(*this, totalCellIdx);
                         }
 
-                    private:
-                        //! Precalulated time-dependent longitudinal value, in this case it is also the final result
-                        float3_X const elong;
-
-                        //! Get time-dependent longitudinal vector factor
-                        HDINLINE float3_X getLongitudinal(float_X const currentStep) const
+                        /** Get time-dependent longitudinal scalar factor for the given time
+                         *
+                         * @param time time moment to calculate the factor at
+                         * @param phaseShift additional phase shift to add on top of everything else,
+                         *                   in radian
+                         */
+                        HDINLINE float_X getLongitudinal(float_X const time, float_X const phaseShift) const
                         {
-                            float_64 const runTime = DELTA_T * currentStep;
-                            float_64 envelope = float_64(Unitless::AMPLITUDE);
-                            float_64 const mue = 0.5 * Unitless::RAMP_INIT * Unitless::PULSE_LENGTH;
-                            float_64 const w = 2.0 * PI * Unitless::f;
-                            float_64 const tau = Unitless::PULSE_LENGTH * math::sqrt(2.0);
-                            float_64 const endUpramp = mue;
-                            float_64 const startDownramp = mue + Unitless::LASER_NOFOCUS_CONSTANT;
-                            float_64 integrationCorrectionFactor = 0.0;
-                            if(runTime > startDownramp)
+                            auto envelope = Unitless::AMPLITUDE;

```

Comment:
In the implementation before we used everywhere 64 bit floating-point numbers. I assume to guarantee stable numerics.
Is the precision change to 32 bit intended?

Reply:
In previous implementation some operations were done on host (e.g. precalculating things for the given `currentStep`). Those were 64-bit. What is in-kernel was 32-bit. Now everything is in-kernel basically, so I converted all values to 32-bit.

Reply:
You have a point tho, that i converted everything more or less on auto pilot. I will go through these changes and try to see if they are potentially dangerous.

Reply:
So i rechecked and in the original (non-incident field) laser profiles all those values were `float_X` except the double-precision `f`. Upon looking at the code, I also see no reason for `f` to require double precision.

Then for the incident field profiles as in the current dev I kinda automatically converted a lot of values to `float_64` as they were only used on the host side for the time-depending part, and only kept those with on-device use `float_X`. However, with this PR due to time depending on position, all the host-side stuff moved back on device and so these values are `float_X` again. This appears safe as, again, almost all were originally `float_X`, and for `f` it also seems fine.